packages/components/src/overlays/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+    composeClassnameRenderProps,
+    SlotProvider,
+    cssModule,
+    ButtonContext,
+    ButtonGroupContext,
+    ContentContext,
+    FooterContext,
+    TextContext,
+    HeadingContext,
+    LinkContext,
+    useColorSchemeContext, HopperProvider, isNil, isFunction
+} from "@hopper-ui/components";
+import { type StyledComponentProps, useStyledSystem } from "@hopper-ui/styled-system";
+import clsx from "clsx";
+import { forwardRef, type ForwardedRef } from "react";
+import {
+    useContextProps,
+    type PopoverProps as RACPopoverProps,
+    type DialogTriggerProps,
+    DialogTrigger,
+    Popover as RACPopover,
+    Dialog
+} from "react-aria-components";
+
+import { PopoverContext } from "./PopoverContext.ts";
+
+import styles from "./Popover.module.css";
+
+export const GlobalPopoverCssSelector = "hop-Popover";
+
+export interface PopoverProps extends StyledComponentProps<RACPopoverProps> {
+}
+
+export interface PopoverTriggerProps extends DialogTriggerProps {
+}
+
+export const PopoverTrigger = (props: PopoverTriggerProps) =>
+    <DialogTrigger {...props}>{props.children}</DialogTrigger>;
+
+function Popover(props: PopoverProps, ref: ForwardedRef<HTMLElement>) {
+    [props, ref] = useContextProps(props, ref, PopoverContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        children,
+        className,
+        offset = 4,
+        ...otherProps
+    } = ownProps;
+
+    const { colorScheme } = useColorSchemeContext();
+
+    const popoverClassNames = composeClassnameRenderProps(
+        className,
+        GlobalPopoverCssSelector,
+        cssModule(
+            styles,
+            "hop-Popover"
+        )
+    );
+
+    return (
+        <RACPopover
+            {...otherProps}
+            offset={offset}
+            ref={ref}
+            className={popoverClassNames}
+        >
+            {state => (
+                <HopperProvider colorScheme={colorScheme}>
+                    <Dialog className={clsx(stylingProps.className, styles["hop-Popover__dialog"])}
+                        style={stylingProps.style}
+                    >
+                        <SlotProvider values={[
+                            [TextContext, {
```

Comment:
Do we have use cases where we want something that is not an heading to be the heading of a popover?

Reply:
This is a difficult question to answer, as we don't yet have any use cases for popovers as components. It will be used as a sub-component of other DS elements.

Reply:
i'm worried that the TextContext like this might affect the Content. 

in theory, something like this: 

```
<Popover>
    <Heading>Title</Heading>
    <Content>
        This content is <Text color="accent">highlighted</Text>
    </Content>
</Popover
```

The Text will have hop-Popover__title and size md passed to it most likely.

Reply:
I think since we have no use case, i'd only let a heading be set, not the Text as a heading

Reply:
no problem, I wonder what happens if someone uses a Text in the heading? Is there any way to block its use?

Reply:
The plan is to avoid using Text content in places that are not directly content, to avoid those issues